### DISCUSSION TITLE
Enun c# console

### DISCUSSION THREAD

**Discussion by RaulCesarM** [community/community#20480] [Other Feature Feedback, Questions, & Ideas]
Como faço para capturar uma opção fornecida por um enum em uma classe porem no console ? tenho uma classe e nela tenho o enum com contendo as cores primarias, mas quando vou escolher qual cor eu quero só posso usar string no readline()..

**Comment by warievilo** [comment]
Você pode fazer da seguinte forma:

`var input = Console.ReadLine();`
`var color = Enum.Parse(typeof(ColorsEnum), input);`

Ou ainda, você pode fazer o cast para inteiro se quiser pegar o valor do enum:

`int value = (int)Enum.Parse(typeof(ColorsEnum), input);`

Obs: O valor digitado pelo usuário deve ser idêntico ao nome do item no enum.
